zerver/lib/test_classes.py

```diff
@@ -184,34 +218,21 @@ def run(self, result: Optional[TestResult]=None) -> Optional[TestResult]:  # noc
     DEFAULT_SUBDOMAIN = "zulip"
     TOKENIZED_NOREPLY_REGEX = settings.TOKENIZED_NOREPLY_EMAIL_ADDRESS.format(token="[a-z0-9_]{24}")
 
+    @staticmethod
+    def _capture_logs_to_string(capture_logs: Any) -> str:
+        return "\n".join(capture_logs.output)
+
     @contextmanager
-    def assertNoLogs(self, test_case,logger_name, level, no_logs, exc_type, exc_value,tb):
-        LOGGING_FORMAT = "%(levelname)s:%(name)s:%(message)s"
-        formatter = logging.Formatter(LOGGING_FORMAT)
-        handler = _CapturingHandler()
-        handler.setLevel(level)
-        handler.setFormatter(formatter)
-        self.watcher = handler.watcher
-        self.old_handlers = logger.handlers[:]
-        self.old_level = logger.level
-        self.old_propagate = logger.propagate
-        logger.handlers = [handler]
-        logger.setLevel(level)
-        logger.propagate = False
-        if self.no_logs:
+    def assertNoLogs(self) -> Any:
+        before_logs = self.captured_logs.output.copy()
+        try:
             yield
-        yield handler.watcher
-        self.logger.handlers = self.old_handlers
-        self.logger.propagate = self.old_propagate
-        self.logger.setLevel(self.old_level)
-        if exc_type is not None:
-            return False
-        if self.no_logs:
-            if len(self.watcher.records) > 0:
-                self._raiseFailure("Unexpected logs found: {!r}".format(self.watcher.output))
-        else:
-            if len(self.watcher.records) == 0:
-                self._raiseFailure("no logs of level {} or higher triggered on {}".format(logging.getLevelName(self.level), self.logger.name))
+        finally:
+            after_logs = self.captured_logs.output
```

Comment:
I think this should give `Attribute Error` if I put this as context manager in any of the test. As `captured_logs` are defined in `LoggingTestCase` class you defined above and you are using them inside `ZulipTestCase` class without extending the 'LoggingTestCase` class.